src/main/scala/com/etsy/conjecture/scalding/util/HyperparameterSearcher.scala

```diff
@@ -0,0 +1,186 @@
+package com.etsy.conjecture.scalding.util
+
+import java.io.Serializable
+
+import cascading.pipe.Pipe
+import com.etsy.conjecture.data._
+import com.etsy.conjecture.evaluation._
+import com.etsy.conjecture.model._
+import com.etsy.conjecture.scalding.evaluate.{BinaryEvaluator, GenericEvaluator, MulticlassEvaluator, RegressionEvaluator}
+import com.etsy.conjecture.scalding.train._
+import com.twitter.scalding.Dsl._
+import com.twitter.scalding._
+
+import scala.util.Random
+
+/**
+  * Samples random parameter values to perform a fast efficient hyperparameter search
+  */
+abstract class HyperparameterSearcher[L <: Label, M <: UpdateableModel[L, M], E <: ModelEvaluation[L]]
+                                      (option: DynamicOptions, parameters: Seq[HyperParameter[_]],
+                                       val numTrials: Int, rng: Random = new Random(0)) extends Serializable {
+
+  val settings = (0 until numTrials).map { trial : Int => trial -> sampledParameters(rng) }.toMap
+
+  def getModelTrainer(args: Args): ModelTrainerStrategy[L, M]
+  val evaluator: GenericEvaluator[L]
+
+  //draw parameter values from given sample method
+  //save values in a new Arg instance
+  def sampledParameters(rng: Random): Args = {
+    parameters.foreach(_.set(rng))
+    option.unParse
+  }
+
+  def search (instances: Pipe, instance_field: Symbol): Pipe = {
+    //Split train test by ratio
+    val splitSet = instances.map(instance_field -> '__fold) { li: LabeledInstance[L] => rng.nextInt(10) <= 7 }
+    val trainSet = splitSet.filter('__fold) { foldId: Boolean => foldId }
+    val testSet = splitSet.filter('__fold) { foldId: Boolean => !foldId }
+
+    val train = generateTrials(trainSet, instance_field)
+    val test = generateTrials(testSet, instance_field)
+
+    val models = trainTrials(train)
+    evaluate(models, test)
+      .groupAll{
+         _.sortBy('eval).reverse
+      }
+  }
+
+  def generateTrials(instances: Pipe, instance_field: Symbol): Pipe = {
+    instances
+      .flatMapTo('instance -> ('instance, 'trial)) {
+         instance: LabeledInstance[L] =>
+             settings.keySet.map {
+               trial: Int =>
+                  (instance, trial)
+             }
+       }
+       .groupBy('trial) {
+          _.toList[LabeledInstance[L]]('instance -> 'instances).reducers(1000)
```

Comment:
This is ok for now but going forward the parallelization should be up to the user with some default, instead of hard coded.